libsoftwarecontainer/component-test/softwarecontainerlib_componenttest.cpp

```diff
@@ -365,9 +367,65 @@ TEST_F(SoftwareContainerApp, TestFolderMounting) {
     job3->start();
     ASSERT_EQ(job3->wait(), EXISTENT);
 
+    // Test that we can mount to some place where there is something else
+    // mounted higher up in the hierarchy (in this case, /dev)
+    ASSERT_TRUE(bindMountInContainer("/dev/shm", "/dev/shm", false));
+    auto jobShm = getSc().createFunctionJob([&] () {
+        return isDirectory("/dev/shm") ? EXISTENT : NON_EXISTENT;
+    });
+    jobShm->start();
+    ASSERT_EQ(jobShm->wait(), EXISTENT);
+}
+
+TEST_F(SoftwareContainerApp, TestPOSIXSHM) {
+    const char *testData1 = "test data to be read";
+    const char *testData2 = "read be to data test";
+    const char *objName1 = "/testObject1";
+    const char *objName2 = "/testObject2";
+
+    // Create a first object
+    int fd = shm_open(objName1, O_CREAT | O_RDWR, 0666);
+    ASSERT_NE(fd, -1);
+    ASSERT_NE(write(fd, testData1, strlen(testData1)), -1);
+    ASSERT_EQ(close(fd), 0);
+
+    // And create a second object
+    fd = shm_open(objName2, O_CREAT | O_RDWR, 0666);
+    ASSERT_NE(fd, -1);
+    ASSERT_NE(write(fd, testData2, strlen(testData2)), -1);
+    ASSERT_EQ(close(fd), 0);
```

Comment:
Consider refactor this to remove code duplication. Not sure what would be best for readability but this is a bit messy...